### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
Only task1 package is related to this PR. Having a single thread that on start requires an end symbol and an value , that it will count till reaching it or reaching the end symbol. The Thread outputs its counting to the console and upon getting to the end count or being interrupted by the end symbol it displays the value it has reached.

### CODE REVIEW

File: src/com/clouway/task1/CountingThread.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.clouway.task1;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class CountingThread implements Runnable {
+  private Integer current = 0;
+  private Integer maxCount;
```
Comment (freekman) [inline src/com/clouway/task1/CountingThread.java]: final

File: src/com/clouway/task1/CountingThread.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  @Override
+  public void run() {
+    while (current <= maxCount) {
+      System.out.println(Thread.currentThread().getName().toString());
+      System.out.println(current);
+      try {
+        Thread.sleep(1000);
+        if (current == maxCount) {
+          System.out.println(current);
+          Thread.interrupted();
```
Comment (freekman) [inline src/com/clouway/task1/CountingThread.java]: ?

File: src/com/clouway/task1/CountingThread.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        current++;
+      } catch (InterruptedException e) {
+        System.out.println(current);
+        break;
```
Comment (freekman) [inline src/com/clouway/task1/CountingThread.java]: why do you break and not return

File: src/com/clouway/task1/CountingThread.java (hunk 2 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task1/CountingThread.java]: A mistake I did its already fixed but am finishing second task b4 pushing

File: src/com/clouway/task1/CountingThread.java (hunk 3 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task1/CountingThread.java]: Changed to return.

File: src/com/clouway/task1/CountingThread.java (hunk 2 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task1/CountingThread.java]: Change visible.

File: src/com/clouway/task1/CountingThread.java (hunk 1 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task1/CountingThread.java]: Made final.

File: src/com/clouway/task1/CountingThread.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  @Override
+  public void run() {
+    while (current <= maxCount) {
+      System.out.println(Thread.currentThread().getName().toString());
+      System.out.println(current);
+      try {
+        Thread.sleep(1000);
+        if (current == maxCount) {
```
Comment (freekman) [inline src/com/clouway/task1/CountingThread.java]: why if current == maxCount do you interupt the current thread

File: src/com/clouway/task2/SynchronyzedThread.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Override
+  public void run() {
+    for (int i = from; i <= to; i++) {
+      System.out.println(Thread.currentThread().getName().toString() + " " + i);
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        System.out.println(Thread.currentThread().getName().toString() + "was interupted at value:");
+        System.out.println(i);
```
Comment (freekman) [inline src/com/clouway/task2/SynchronyzedThread.java]: ?

File: src/com/clouway/task1/CountingThread.java (hunk 4 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task1/CountingThread.java]: Task description. The thread should count till reaching the given value.

File: src/com/clouway/task2/SynchronyzedThread.java (hunk 5 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task2/SynchronyzedThread.java]: Some feedback for me. I could remove it, shorten it or whatever is necessary.

File: src/com/clouway/task3/SynchronyzedThread.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  @Override
+  public void run() {
+    synchronized (this) {
+      for (int i = from; i <= to; i++) {
+        System.out.println(Thread.currentThread().getName().toString() + " " + i);
+        System.out.println();
+        try {
+          Thread.sleep(1000);
```
Comment (freekman) [inline src/com/clouway/task3/SynchronyzedThread.java]: tooo complicated 
for this task take a look at Synchronized Methods

File: src/com/clouway/task5/Killer.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class Killer extends Thread {
+  private TimeoutHashtable timeoutHashtable;
+  private Integer ttc;
+
+  /**
+   * A Killer Thread that iterates through the Map to check for dead values.
```
Comment (freekman) [inline src/com/clouway/task5/Killer.java]: why is there so much boiler plate comments

File: src/com/clouway/task5/TimeoutHashtable.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class TimeoutHashtable {
+
+  private Integer ttl;
+
+  /**
+   * A concurrent HashMap containing Integer Keys and TTLTHreads that live a given period of time.
```
Comment (freekman) [inline src/com/clouway/task5/TimeoutHashtable.java]: .

File: src/com/clouway/task5/TimeoutHashtable.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+      ttlThread.start();
+    }
+  }
+
+  /**
+   * Removes a thread from the Map.
+   * @param key The key from the map.
+   * @return Returns the key that is removed, or null if there is no such key in it.
+   */
+  public synchronized Integer remove(Integer key) {
```
Comment (freekman) [inline src/com/clouway/task5/TimeoutHashtable.java]: why does remove return an Integer

**Comment by freekman** [comment]
LGTM after fix comments

File: src/com/clouway/task5/TimeoutHashtable.java (hunk 9 above)
Comment (VasilMitovClouway) [inline src/com/clouway/task5/TimeoutHashtable.java]: As a task requirement: remove(K key) - маха елемент по ключ, ако го няма връща null.

Comment (freekman) [inline src/com/clouway/task5/TimeoutHashtable.java]: ok

**Pull request merged**